q2cli/util.py

```diff
@@ -73,6 +77,43 @@ def exit_with_error(e, header='An error has been encountered:',
     click.get_current_context().exit(status)
 
 
+def output_in_cache(fp):
+    """Determines if an output path follows the format
+    /path_to_extant_cache:key
+    """
+    from pathlib import Path
+    from qiime2.core.cache import Cache
+
+    # Tells us right away this isn't in a cache
+    if ':' not in fp:
+        return False
+
+    split_path = fp.split(':')
+
+    # Account for potential for : in the path
+    cache_path = ':'.join(elem for elem in split_path[:-1])
+    key = split_path[-1]
+
+    try:
+        if Cache.is_cache(Path(cache_path)):
+            if not key.isidentifier():
+                raise ValueError(
+                    'Key must be a valid Python identifier. Python identifier '
+                    'rules may be found here https://www.askpython.com/python/'
+                    'python-identifiers-rules-best-practices')
+            else:
+                return True
+    except FileNotFoundError as e:
+        # If cache_path doesn't exist, don't treat this as a cache output
+        if 'No such file or directory' in str(e):
+            pass
+        else:
+            raise e
```

Comment:
Is this branch reachable given the specific `FileNotFoundError` filter?

Reply:
Probably not. I wanted to be certain that if some other error did somehow get raised like that we didn't just squash it. Should I just remove the else?